app/src/main/kotlin/com/mapzen/erasermap/controller/MainActivity.kt

```diff
@@ -436,6 +438,13 @@ class MainActivity : AppCompatActivity(), MainViewController, RouteCallback,
         val cacheSearches = PreferenceManager.getDefaultSharedPreferences(this)
                 .getBoolean(AndroidAppSettings.KEY_CACHE_SEARCH_HISTORY, true)
         searchView?.setCacheSearchResults(cacheSearches)
+        val closeButton = searchView?.findViewById(R.id.search_close_btn) as ImageView
+        closeButton.setOnClickListener {
```

Comment:
I like the first suggestion to hide the close button, I think it feels most natural. Good call, I like it better than overriding the click listener!